### ISSUE TITLE
libdbd-sqlite3-perl: FTBFS (test failure in t/57_uri_filename.t) with sqlite3 URI filename support

### ISSUE THREAD

**Issue by gregoa** [DBD-SQLite/DBD-SQLite#47]
We have the following bug reported to the Debian package of
DBD-SQLite (https://bugs.debian.org/922248):

It doesn't seem to be a bug in the packaging (module the fact that we are
building against the system sqlite3), so you may want to take a look. 
Thanks!

------8<-----------8<-----------8<-----------8<-----------8<-----

```
Source: libdbd-sqlite3-perl
Version: 1.60-1
Severity: serious
Tags: upstream ftbfs sid
Justification: fails to build from source (but built successfully in the past)

-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

libdbd-sqlite3-perl (both 1.60-1 in testing as 1.62-1 in unstable)
fail to build (currently only in unstable) against the new sqlite3
3.27.1-1:

t/57_uri_filename.t ...................................
1..17
not ok 1 - correct database is not created for uri
ok 2 - correct database is created for uri
ok 3 - failed to open a nonexistent readonly database for uri
ok 4 - failed to open a nonexistent readwrite database for uri
ok 5 - correct database is created for uri
ok 6 - created a dbfile
ok 7 - opened a correct readonly database for uri
ok 8 - created a dbfile
ok 9 - opened a correct readwrite database for uri
ok 10 - correct database is created for uri
ok 11 - failed to open a nonexistent readonly database for uri
ok 12 - failed to open a nonexistent readwrite database for uri
ok 13 - correct database is created for uri
ok 14 - created a dbfile
ok 15 - opened a correct readonly database for uri
ok 16 - created a dbfile
ok 17 - opened a correct readwrite database for uri
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/17 subtests



When looking at sqlite3's changelog, I guess this is from

  * Compile with URI filename support.

as the test does

    DBI->connect("dbi:SQLite:$uri{base}", undef, undef, {

(while the next, passing test does

    DBI->connect("dbi:SQLite:uri=$uri{base}", undef, undef, {

)



Currently this affects only unstable, and as long as sqlite3 doesn't
migrate (it already has a autopkgtest regression warning about
breaking libdbd-sqlite3-perl) that's fine. If sqlite3 migrates we
have a problem because even if we disable the test or upstream comes
up with a different fix, 1.62-1 doesn't build on mips (#919773).


Cheers,
gregor

-----BEGIN PGP SIGNATURE-----

iQKTBAEBCgB9FiEE0eExbpOnYKgQTYX6uzpoAYZJqgYFAlxkXVdfFIAAAAAALgAo
aXNzdWVyLWZwckBub3RhdGlvbnMub3BlbnBncC5maWZ0aGhvcnNlbWFuLm5ldEQx
RTEzMTZFOTNBNzYwQTgxMDREODVGQUJCM0E2ODAxODY0OUFBMDYACgkQuzpoAYZJ
qgYvxQ//apJ4vy9O2pZHMA+TfIl9rYecOiQ3p1n1Z/xnDjUgCgKNKzrOESRTV/HX
9l78V/R63KlbKlwofmkUNCfJZ8dKR8YAOc8JXImDv/Aflh54MsrTxXc+KJLEFhcb
K4nen1k1Pxb2hzpxZWktUlZCxTFJ32p6weFgmLntEiXmktR1mC2b1H8fwsJ1O06h
aUwyGVzszmFfM5fzSUY2mYZTW3YiCGSpTB8GXCfUQFDTI4hPh1i8GsCjOWndOr1V
ZJGpmgrPFgQb/s0qj5ZN8XSAfo4tnma0r91wVC00vL4tm6vgvzVa3q+EkDKrcZ9o
6kCwruA3lkUIq2AQM18pubQqcK30IcwpsJUzQ+K+1tzlWv3Hm0MVJa5VG12zdhmK
avyYRSD+a+gRJFtOhruTkOiRfcAvu0i3BT4Beklcg6Jmt8awZvfz0aZUwGTu2mwD
/CHUvNv8/Cc+AbxMLGnibWVPVf5i3+1QOreRb5fhBUMYF5PY/jK1U2+8gaSKRcXj
Gs/Vluz0pmK2Y3AiQZV8jCaPQ+nfAe138tb/gB6WvWC+QNNrprxiZYSfbSs3DRLy
1vhJI3BsB4f9NEuj4x7U8oHpc0DQFMTZLtSU6Vn6P4SqmGKEeqUvEuKrQkxMHuDY
ckgbssVoagDmIlWndHagyEVwtIVlj5W+z70LyLbtmuJT6OFq/po=
=9KyR
-----END PGP SIGNATURE-----


```
------8<-----------8<-----------8<-----------8<-----------8<-----

Thanks for considering,
  gregor herrmann,
  Debian Perl Group

**Comment by charsbar** [comment]
Thanks for the heads-up. The commit above should fix this issue (not yet released).

**Comment by gregoa** [comment]
Thanks very much. This does indeed look good.

Alas …
```
t/57_uri_filename.t .. 
1..17
Undefined subroutine &main::has_compile_option called at t/57_uri_filename.t line 38.
# Looks like your test exited with 2 before it could output anything.
Dubious, test returned 2 (wstat 512, 0x200)
Failed 17/17 subtests 
```

Ah, this needs a newer ```t/lib/SQLiteTest.pm``` from f56689d929b8b00366e56439be1340c31c145825 …

*some minutes later*

Yay, works :)
Thanks again. (And feel free to close this iusse.)

**Comment by charsbar** [comment]
Shipped 1.63_02 with this. Thanks!
